solr/core/src/test/org/apache/solr/handler/admin/api/ReplaceNodeAPITest.java

```diff
@@ -0,0 +1,85 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.handler.admin.api;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.cloud.OverseerSolrResponse;
+import org.apache.solr.cloud.api.collections.DistributedCollectionConfigSetCommandRunner;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.core.CoreContainer;
+import org.apache.solr.handler.admin.CollectionsHandler;
+import org.apache.solr.jersey.SolrJerseyResponse;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/** Unit tests for {@link ReplaceNodeAPI} */
+public class ReplaceNodeAPITest extends SolrTestCaseJ4 {
+
+  private CoreContainer mockCoreContainer;
+  private static final String sourceNodeName = "demoSourceNode";
+  private static final String targetNodeName = "demoTargetNode";
+  public static final String async = "async";
+  private static final boolean waitForFinalState = false;
+  private SolrQueryRequest mockQueryRequest;
+  private SolrQueryResponse queryResponse;
+  private ReplaceNodeAPI replaceNodeApi;
+  private DistributedCollectionConfigSetCommandRunner mockCommandRunner;
+
+  @BeforeClass
+  public static void ensureWorkingMockito() {
+    assumeWorkingMockito();
+  }
+
+  @Override
+  @Before
+  public void setUp() throws Exception {
+    super.setUp();
+
+    mockCoreContainer = mock(CoreContainer.class);
+    mockCommandRunner = mock(DistributedCollectionConfigSetCommandRunner.class);
+    when(mockCoreContainer.getDistributedCollectionCommandRunner())
+        .thenReturn(Optional.of(mockCommandRunner));
+    when(mockCommandRunner.runCollectionCommand(any(), any(), anyLong()))
+        .thenReturn(new OverseerSolrResponse(new NamedList<>()));
+    mockQueryRequest = mock(SolrQueryRequest.class);
+    queryResponse = new SolrQueryResponse();
+    replaceNodeApi = new ReplaceNodeAPI(mockCoreContainer, mockQueryRequest, queryResponse);
+  }
+
+  @Test
+  public void testSuccessfulReplaceNodeCommand() throws Exception {
+    when(mockCoreContainer.isZooKeeperAware()).thenReturn(true);
+    final String expectedJson = "{\"responseHeader\":{\"status\":0,\"QTime\":0}}";
+    final CollectionsHandler collectionsHandler = mock(CollectionsHandler.class);
+    when(mockCoreContainer.getCollectionsHandler()).thenReturn(collectionsHandler);
```

Comment:
[0] I suspect this line and the one above it are leftover from earlier testing?  At the least they're not doing anything right now - I double-checked ReplaceNodeAPI and it never calls `getCollectionsHandler` or anything similar.

Similarly, the `expectedJson` variable isn't used afaict.